app/api/views/briefs.py

```diff
@@ -1183,102 +1155,189 @@ def download_brief_response_file(brief_id, supplier_code, slug):
+        audit_service.log_audit_event(
+            audit_type=AuditTypes.create_brief_response,
+            user=current_user.email_address,
+            data={
+                'briefResponseId': brief_response.id
+            },
+            db_object=brief_response
+        )
+        publish_tasks.brief_response.delay(
+            publish_tasks.compress_brief_response(brief_response),
+            'created',
+            user=current_user.email_address
+        )
     except Exception as e:
-        brief_response_json['brief_id'] = brief_id
-        rollbar.report_exc_info(extra_data=brief_response_json)
+        rollbar.report_exc_info()
 
-    audit_service.log_audit_event(
-        audit_type=AuditTypes.create_brief_response,
-        user=current_user.email_address,
-        data={
-            'briefResponseId': brief_response.id,
-            'briefResponseJson': brief_response_json,
-        },
-        db_object=brief_response)
+    return jsonify(brief_response.serialize()), 201
 
-    publish_tasks.brief_response.delay(
-        publish_tasks.compress_brief_response(brief_response),
-        'submitted',
-        user=current_user.email_address
-    )
-    return jsonify(briefResponses=brief_response.serialize()), 201
+
+@api.route('/brief/<int:brief_id>/respond/<int:brief_response_id>', methods=['PATCH'])
+@login_required
+@role_required('supplier')
+def update_brief_response(brief_id, brief_response_id):
+    brief_response_json = get_json_from_request()
+    supplier, brief = _can_do_brief_response(brief_id, update_only=True)
+    brief_response = brief_responses_service.find(
+        id=brief_response_id,
+        brief_id=brief.id,
+        supplier_code=supplier.code,
+        withdrawn_at=None
+    ).one_or_none()
+    if not brief_response or brief_response.status not in ['submitted', 'draft']:
+        not_found('Brief response not found')
+    if brief.status != 'live':
+        abort('Brief responses can only be edited when the brief is still live')
+
+    submit = False
+    if 'submit' in brief_response_json:
+        if brief_response_json['submit']:
+            submit = True
+        del brief_response_json['submit']
+
+    # if the current brief response is already submitted and it doesn't have the required upload file fields, then
+    # flag the required file check as unnecessary as this would be an old response before the introduction of the
+    # individual file fields in the brief response
+    do_required_file_check = True
+    if brief_response.status == 'submitted' and (
+        len(brief_response.data.get('resume', [])) == 0 and
+        len(brief_response.data.get('responseTemplate', [])) == 0 and
+        len(brief_response.data.get('writtenProposal', [])) == 0
+    ):
+        do_required_file_check = False
+
+    brief_response.data = brief_response_json
+    if submit:
+        try:
+            brief_response.validate(do_required_file_check=do_required_file_check)
```

Comment:
Rather than validate on the model, should we call a function that lives in a file under `/api/business/validators`?

Reply:
This is a good idea, but perhaps as a separate task/PR? only minor changes were made to the brief response validation in this PR, and refactoring so it's similar to how we do validation on briefs would be a fairly major change which I'd argue is not necessary for this PR (this whole PATCH endpoint is really just the same as the existing brief response POST with a few relevant tweaks for differentiating between saves and submits).